lib/proxy/pluggable.rb

```diff
@@ -1,131 +1,170 @@
+    @custom_validators = hash_of_validators
   end
 
-  module ClassMethods
-    attr_reader :plugin_name, :version, :after_activation_blk, :bundler_group_name
+  def load_dependency_injection_wirings(class_name_to_use = nil, &block_to_use)
+    @di_wirings_loader = class_name_to_use || block_to_use
+  end
 
-    # Methods below define DSL for defining plugins
+  def load_programmable_settings(class_name_to_use = nil, &block_to_use)
+    @programmable_settings = class_name_to_use || block_to_use
+  end
 
-    def after_activation(&blk)
-      @after_activation_blk = blk
-    end
+  def load_classes(class_name_to_use = nil, &block_to_use)
+    @class_loader = class_name_to_use || block_to_use
+  end
 
-    def requires(plugin_name, version_spec)
-      self.dependencies << ::Proxy::Dependency.new(plugin_name, cleanup_version(version_spec))
-    end
+  def start_services(*di_labels)
+    @services = di_labels
+  end
 
-    def bundler_group(name)
-      @bundler_group_name = name
-    end
+  #
+  # End of DSL
+  #
 
-    # relative to ::Proxy::SETTINGS.settings_directory
-    def settings_file(apath = nil)
-      if apath.nil?
-        @settings_file || "#{plugin_name}.yml"
-      else
-        @settings = nil
-        @settings_file = apath
-      end
-    end
+  def settings
+    @settings ||= Proxy::Settings.load_plugin_settings(plugin_default_settings, settings_file)
+  end
 
-    def default_settings(a_hash = {})
-      @settings = nil
-      @plugin_default_settings ||= {}
-      @plugin_default_settings.merge!(a_hash)
-    end
+  def settings=(arg)
+    @settings = arg
+  end
 
-    def initialize_after(*module_names)
-      raise "#{plugin_name}: 'initialize_after' method has been removed."
-    end
+  def module_loader_class
+    @module_loader_class ||= after_activation_blk.nil? ? ::Proxy::DefaultModuleLoader : ::Proxy::LegacyModuleLoader
+  end
 
-    def validate_readable(*settings)
-      # Passing in plugin class and setting name is a bit awkward, but we need to delay the loading of module settings
-      # until after module's Plugin/Provider class has been loaded (to preserve order-independence of statements used in
-      # the class body of the Plugin.)
-      settings.each { |setting| validators << ::Proxy::PluginValidators::FileReadable.new(self, setting) }
+  class ClassLoaderProcWrapper
+    def initialize(a_blk); @a_blk = a_blk; end
+    def load_classes; @a_blk.call; end
+  end
+  def class_loader
+    return nil if @class_loader.nil?
+    case @class_loader
+    when String
+      eval(@class_loader).new
+    when Proc
+      ClassLoaderProcWrapper.new(@class_loader)
+    else
+      @class_loader.new
     end
+  end
 
-    def validate_presence(*settings)
-      settings.each { |setting| validators << ::Proxy::PluginValidators::Presence.new(self, setting) }
+  class SettingsProcWrapper
+    def initialize(a_blk); @a_blk = a_blk; end
+    def load_programmable_settings(settings); @a_blk.call(settings); end
```

Comment:
Can you expand these out rather than using semicolons?  It's more readable - also in classes above and below.